control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs

```diff
@@ -130,49 +134,247 @@ async fn nexus_reconciler(
+                                            )),
+                                        )
+                                        .await?;
+                                } else {
+                                    let child_node =
+                                        get_child_node(nexus_spec.clone(), ch, context).await?;
+                                    let node = context.registry().node_wrapper(&child_node).await?;
+                                    let node_status = node.read().await.status();
+                                    match node_status {
+                                        NodeStatus::Unknown | NodeStatus::Offline => {
+                                            info!("Replica node is still not online");
+                                        }
+                                        NodeStatus::Online => {
+                                            info!("Child node came back {:?}", child_node);
+                                            let child_replica =
+                                                get_child_replica(nexus_spec.clone(), ch, context)
+                                                    .await?;
+                                            info!("child replica object is : {:?}", child_replica);
+                                            let rep_pool = child_replica.pool_id;
+                                            let pool =
+                                                context.registry().get_pool(&rep_pool).await?;
+                                            if let Some(state) = pool.state() {
+                                                if state.status == PoolStatus::Online {
+                                                    info!("Child pool is imported and its online");
+                                                    nexus
+                                                        .set_rebuild_state(
+                                                            context.registry(),
+                                                            ch.uri.clone(),
+                                                            Some(RebuildInfo::new(
+                                                                None,
+                                                                RebuildStage::PartialRebuildInit,
+                                                            )),
+                                                        )
+                                                        .await?;
+                                                }
+                                            } else {
+                                                info!("Child pool is not imported yet");
+                                            }
+                                        }
+                                    };
+                                }
+                            }
+                        }
```

Comment:
we can simplify this by simply checking if we have the replica in the registry.

Reply:
I think i was getting replica from the registry even when node is Offline. Thats how i get child node. You want me to check status/state of the Replica?